recipes/nccl/build.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+################################################
+# Use `asm-generic/socket.h` from distro.      #
+# Needed to ensure `SO_REUSEPORT` is defined.  #
+# This feature was added in Linux kernel 3.9.  #
+# However RHEL 6 and CentOS 6 backported it.   #
+# The sysroot compilers get in the way so we   #
+# use this header from this system to define   #
+# `SO_REUSEPORT`.                              #
+#                                              #
+# ref: https://lwn.net/Articles/542629/        #
+################################################
+CONDA_BUILD_SYSROOT="$(${CC} --print-sysroot)"
+cp /usr/include/asm-generic/socket.h "${CONDA_BUILD_SYSROOT}/usr/include/asm-generic/socket.h"
```

Comment:
To make this package buildable everywhere, it's better to define `-DSO_REUSEPORT=<value>` in CPPFLAGS.

Reply:
Please raise an issue in the feedstock and we can discuss.